src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private static final double DEFAULT_SIZE_GROW = 1.5;
+
+    private int size;
+    private T[] values;
+
+    public ArrayList() {
+        values = (T[]) new Object[MAX_ITEMS_NUMBER];
+    }
+
     @Override
     public void add(T value) {
-
+        checkSize();
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index != size) {
+            checkIndex(index);
+        }
+        checkSize();
+        T[] resultCopy = (T[]) new Object[values.length];
+        System.arraycopy(values, 0, resultCopy, 0, index + 1);
+        resultCopy[index] = value;
+        System.arraycopy(values, index, resultCopy, index + 1, size - index);
+        values = resultCopy;
+        size++;
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return values[index];
     }
 
     @Override
     public void set(T value, int index) {
+        checkIndex(index);
+        values[index] = value;
 
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+
+        final T removeValue = (T) values[index];
+        final T[] resultCopy = (T[]) new Object[values.length - 1];
+        System.arraycopy(values, 0, resultCopy, 0, index);
+        System.arraycopy(values, index + 1, resultCopy, index, size - (index + 1));
+        values = resultCopy;
```

Comment:
Too many heavy operations for this logic.
You should use 1 arraycopy() method to move tail of array 1 position to the left (you can copy and insert in same array)
e.g.
was [0, 1, 2, 3, 4], we deleting 2
we copying 3 and 4 for one position to the left, obtaining [0, 1, 3, 4, 4]
and then we reducing size variable which cuts unimportant tail: [0, 1, 3, 4]